lms/djangoapps/instructor/services.py

```diff
@@ -78,13 +76,60 @@ def delete_student_attempt(self, student_identifier, course_id, content_id, requ
             except (StudentModule.DoesNotExist, enrollment.sub_api.SubmissionError):
                 err_msg = (
                     'Error occurred while attempting to reset student attempts for user '
-                    '{student_identifier} for content_id {content_id}.'.format(
-                        student_identifier=student_identifier,
-                        content_id=content_id
-                    )
+                    f'{student_identifier} for content_id {content_id}.'
                 )
                 log.error(err_msg)
 
+    def complete_student_attempt(self, user_identifier, content_id):
+        """
+        """
+        try:
+            user = get_user_by_username_or_email(user_identifier)
+        except ObjectDoesNotExist:
+            err_msg = (
+                'Error occurred while attempting to complete student attempt for user '
+                f'{user_identifier} for content_id {content_id}. '
+                'User does not exist!'
+            )
+            log.error(err_msg)
+            return
+
+        try:
+            block_key = UsageKey.from_string(content_id)
+        except InvalidKeyError:
```

Comment:
not sure which is more pythonic, though i think you could also put the user = and block_key = lines together, and then have the exceptions follow one after the other, rather than having two separate try/except blocks
to me having a single try with multiple except might be clearer, but not sure, maybe this is better

Reply:
Ah hmm, I'll maybe play with it and see which I prefer. I like the explicitness here so you very clearly know which exception relates to which piece of code, but with those lines, very likely to be clear anyway.